compiler/rustc_attr_parsing/src/attributes/mod.rs

```diff
@@ -231,7 +231,10 @@ pub(crate) enum AttributeOrder {
     KeepLast,
 }
 
-type ConvertFn<E> = fn(ThinVec<E>) -> AttributeKind;
+pub(crate) enum ConvertFn<E> {
```

Comment:
I've changed `ConvertFn` to an enum so I can also keep track of the span of the first attribute. I use this span to lint on the attribute itselfs (rather than one of the elements) for a few of the diagnostics. Does this seem reasonsable?

Reply:
I redefined `ConvertFn` as `fn(ThinVec<E>, Span) -> AttributeKind;`. I think this is what you meant, right? 
I don't think we need the `DUMMY_SP` think you were talking about, as `ConvertFn` is only called when there is at least one attribute, in which case there is always a real span.